test/hw_tests/hwtest_timestamp_standalone.py

```diff
@@ -0,0 +1,153 @@
+    try:
+        return UdpData(*[read_hex_value(line, key) for key in UDP_DATA_HEX_KEYS],
+                       read_number(line, "time_epoch") * 1000000000)
+    except AttributeError:
+        return []
+
+
+def read_udp_pakets(data: str):
+    """ Lines which do not contain data for all keys are omitted.
+    """
+    return [read_udp_line(line) for line in data.splitlines()]
+
+
+def read_timestamp_output(data: str):
+    stamps = []
+    for l in data.splitlines():
+        with contextlib.suppress(ValueError):
+            stamps.append(read_timestamp(l))
+    return stamps
+
+
+def read_timestamp(line: str):
+    l = line.split(": ")
+    return TimeStamp(int(l[2]), int(l[3]))
+
+
+def run_command(command, timeout, **kwargs):
+    print("\n\nExecuting: %s\n" % (command))
+    start = time.time()
+    full_output = ""
+    with subprocess.Popen(command, stdout=subprocess.PIPE, shell=True, **kwargs) as process:
+        while (time.time() - start) < timeout:
+            output = process.stdout.readline().decode()
+            if output:
+                full_output += output
+        process.kill()
+    return full_output.strip()
+
+
+def pair_udp_data_and_stamps(udp_data: Sequence[UdpData], stamps: Sequence[TimeStamp]):
+    pairs = []
+    for t in stamps:
+        first = None
+        last = None
+        for u in udp_data:
+            if t.scan_counter == u.scan_counter:
+                if not first or first.from_theta > u.from_theta:
+                    first = u
+                if not last or last.from_theta < u.from_theta:
+                    last = u
+        if first and last:
+            pairs.append(Pair(first, last, t))
+    return pairs
+
+
+class HwtestTimestampStandalone(unittest.TestCase):
+    """ ToDo
+    """
+
+    def setUp(self):
+        self.udp_recorder_thread = Thread(target=self.record_udp_pakets)
+        self.udp_recorder_thread.start()
+        self.scanner_driver_thread = Thread(target=self.run_scanner_driver)
+        self.scanner_driver_thread.start()
+
+    def tearDown(self):
+        self.udp_recorder_thread.join()
+
+    def record_udp_pakets(self):
+        self.udp_pakets = run_command(
+            f"sh {Path(__file__).parent.parent/Path('scripts/record_udp_pakets.sh')} {WAIT_TIME_S} {UDP_DATA_PORT}", WAIT_TIME_S)
+
+    def run_scanner_driver(self):
+        self.timestamp_output = run_command(
+            f"rosrun psen_scan_v2 timestamp_node timeout {WAIT_TIME_S}", WAIT_TIME_S)
+
+    def wait_for_msgs(self):
+        self.scanner_driver_thread.join()
+        self.udp_recorder_thread.join()
+
+    def test_timestamp(self):
+        self.wait_for_msgs()
+        print("Done Captureing\n")
+        lines = read_udp_pakets(self.udp_pakets)
+        stamps = read_timestamp_output(self.timestamp_output)
+        print(f"have {len(lines)} packages and {len(stamps)} callback calls")
+
+        pairs = pair_udp_data_and_stamps(lines, stamps)
+        for last_sr, sr in zip(pairs, pairs[1:]):
```

Comment:
If you use it like that, `pairs` should be ordered with respect to the scan counters. (Which it probably is, but we need to be shure)